gdal/ogr/ogrsf_frmts/cad/gdalcaddataset.cpp

```diff
@@ -0,0 +1,463 @@
+/*******************************************************************************
+ *  Project: OGR CAD Driver
+ *  Purpose: Implements driver based on libopencad
+ *  Author: Alexandr Borzykh, mush3d at gmail.com
+ *  Author: Dmitry Baryshnikov, polimax@mail.ru
+ *  Language: C++
+ *******************************************************************************
+ *  The MIT License (MIT)
+ *
+ *  Copyright (c) 2016 Alexandr Borzykh
+ *  Copyright (c) 2016, NextGIS
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *******************************************************************************/
+#include "ogr_cad.h"
+#include "cpl_conv.h"
+#include "gdal_proxy.h"
+#include "gdal_pam.h"
```

Comment:
I suggest this headers to look like:
#include "ogr_cad.h"
#include "vsilfileio.h"
// blank line separates gdal headers and locals
#include "cpl_conv.h"
#include "gdal_pam.h"
#include "gdal_proxy.h"